api/types/authentication.go

```diff
@@ -535,6 +536,72 @@ func (u *U2F) Check() error {
 			return trace.BadParameter("u2f configuration has an invalid attestation CA: %v", err)
 		}
 	}
+
+	// Removes the port from a URL host.
+	getDomain := func(s string) (string, error) {
+		if strings.Contains(s, ":") {
+			host, _, err := net.SplitHostPort(s)
+			if err != nil {
+				return "", trace.Wrap(err)
+			}
+
+			return host, nil
+		}
+
+		return s, nil
+	}
+
+	// AppID must be a valid URL.
+	appIDUrl, err := url.Parse(u.AppID)
```

Comment:
Is this gonna work when you skip the schema ex `www.foo.com`? I assume that this case will fail on some other validation, as it should, but explicit check would probably make debugging easier. I remember that `url.Parse()` is doing something weird then the passed string is not a valid URL.

Reply:
This doesn't always fail when that is the case. Only if a facet has a scheme when the appid doesnt. The reason for this is that we rely on this behaviour to derive webauthn configs from some u2f configs so we allow the check to pass if the facets are also naked.

Reply:
I added a comment to clarify.